yaksh/models.py

```diff
@@ -1197,7 +1215,18 @@ def make_answerpaper(self, user, ip, attempt_num, course_id):
             ans_paper.save()
             questions = self._get_questions_for_answerpaper()
             ans_paper.questions.add(*questions)
-            question_ids = [str(que.id) for que in questions]
+            question_ids = []
+            for question in questions:
+                question_ids.append(str(question.id))
+                testcases = question.get_test_cases()
+                if self.shuffle_testcases and \
+                    question.type in ["mcq", "mcc"]:
+                    random.shuffle(testcases)
+                testcases_ids = ",".join([str(tc.id) for tc in testcases])
+                testcases_order = TestCaseOrder.objects.create(
```

Comment:
I think what Sir pointed out here is why TestCaseOrder object is created for every questions! Since order is only needed for MCQ and MCC, so in TestCaseOrder, the order should be saved only for these question types.